shrine_images/lib/shrine_images.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Needed to get path of this package.
+import 'dart:mirrors';
+import 'dart:io';
+
+class ShrineImages {
+  List<String> get imageNames => List<String>.unmodifiable(_imagenames);
+
+  ShrineImage operator [](String name) {
+    if (!_imagenames.contains(name)) {
+      throw Exception('Unknown image "$name"');
+    }
+    return ShrineImage._(name);
+  }
+}
+
+class ShrineImage {
+  ShrineImage._(this._imagename);
+  final Uri baseUri = currentMirrorSystem().libraries['shrine_images'].uri;
```

Comment:
Instead of mirrors can you just use Platform.script or maybe Platform.script.path?  Though I'm not sure that will work on all platforms.

You could also just have a method for the web uri and one for getting the mobile device uri if that doesn't work.

Reply:
Thanks for the input Dan. I've decided I can get a more interesting demo on stage if I actually implement a small client library for [Unsplash](https://unsplash.com/)'s API surface instead. It allows me to talk about hiding an API key in a server, exposing a use case specific API to the Flutter client, and sharing code between the server and client in a way that doesn't upset Docker. Oh, and making a small library package for the API client itself, just for completeness.  =)